views/timeslot.tmpl

```diff
@@ -19,16 +19,35 @@
 {{define "content"}}
 <div class="container-fluid current-next">
   <div class="container">
-  {{if .PageData.MixcloudAvailable}}
+  {{if ne .PageData.ODState 0}}
     <div class="row justify-content-center">
-      <div class="col-11 col-md-10 py-2 mixcloud">
+      <div class="col-11 col-md-10 py-2 mixcloud text-center text-lg-left">
         <h2>
           <a href="/ontap/" title="Goto {{.PageContext.ShortName}} {{.PageContext.ODName}}">
             {{- .PageContext.ShortName}} {{.PageContext.ODName -}}
-          </a>: Listen Again
+          </a>
+          {{- if ge .PageData.ODState 3 -}}: Listen Again{{end}}
         </h2>
-        <iframe width="100%" height="120" src="//www.mixcloud.com/widget/iframe/?feed={{.PageData.Timeslot.MixcloudStatus}}&hide_cover=1&light=1" frameborder="0"></iframe>
-        <a href="/ontap/" title="Goto {{.PageContext.ShortName}} {{.PageContext.ODName}}">Listen to more shows on demand...</a>
+        {{if eq .PageData.ODState 1}}
+          <h3>This show starts on {{.PageData.Timeslot.StartTime.Format "Monday, _2 Jan 2006 at 15:04"}}.</h3>
+          <p class="h5">Just can't wait? Check out previous shows <a href="{{.PageData.Timeslot.Season.ShowMeta.MicroSiteLink.URL}}">{{.PageContext.ODName -}}...</a></p>
+        {{else if eq .PageData.ODState 2}}
+          <h3 class="float-lg-left">This show is on right now!</h3>
+            <a href="https://ury.org.uk/listen" class="btn btn-xl btn-red listen-btn float-lg-right" title="Listen Live" onclick="window.open('http://ury.org.uk/live', 'radioplayer', 'height=665,width=380'); return false;">
+              Listen Live
+            </a>
+          <br><br>
+          <p class="float-lg-left h5">Missed the start? Check out previous shows while you wait, <a href="{{.PageData.Timeslot.Season.ShowMeta.MicroSiteLink.URL}}">{{.PageContext.ODName}}...</a></p>
+        {{else if eq .PageData.ODState 3}}
+          <h3>This show is being processed, it should be available soon!</h3>
+          <p class="h5">Just can't wait? Check out previous shows <a href="{{.PageData.Timeslot.Season.ShowMeta.MicroSiteLink.URL}}">{{.PageContext.ODName -}}...</a></p>
+        {{else if eq .PageData.ODState 4}}
+          <iframe width="100%" height="120" src="//www.mixcloud.com/widget/iframe/?feed={{.PageData.Timeslot.MixcloudStatus}}&hide_cover=1&light=1" frameborder="0"></iframe>
+          <a href="/ontap/" title="Goto {{.PageContext.ShortName}} {{.PageContext.ODName}}">Listen to more shows on demand...</a>
+        {{else}}
+          <h3>Sorry, this show isn't available for catchup.</h3>
+          <p class="h5">No recording for this episode is available, this may be at the presenter's request.</p>
```

Comment:
"this may be at the presenter's request" still reads weirdly to me, i'd just get rid of it

Reply:
hmm